rs-client-demo/src/main.rs

```diff
@@ -22,71 +24,92 @@ pub struct LatestChangeEntry {
     pub collection: String,
 }
 
-#[tokio::main]
+#[tokio::main(flavor = "current_thread")]
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let server_url = option_env!("SERVER_URL").unwrap_or("http://localhost:8888/v1");
     let authorization = option_env!("AUTHORIZATION").unwrap_or("");
-    // The `product-integrity` collection is automatically created in CI because
-    // we use the `testing.ini` config which will create all signed resources on startup.
-    // See `KINTO_SIGNER_RESOURCES` in `.circleci/config.yml`.
-    let collection = option_env!("COLLECTION").unwrap_or("product-integrity");
+    let env_name = option_env!("ENV").unwrap_or("prod").to_lowercase();
 
     env_logger::init();
     set_backend(&ReqwestBackend).unwrap();
 
-    println!("Connect to local server {}", server_url);
-    let editor_client = Client::builder()
-        .http_client(Box::new(ViaductClient))
-        .server_url(server_url)
-        .authorization(authorization)
-        .collection_name(collection)
-        .build()
-        .unwrap();
-
-    println!("main-workspace/{}: Create a record", collection);
-    editor_client
-        .store_record(Record::new(json!({
-          "id": "my-key",
-          "foo": "bar"
-        })))
-        .await?;
-
-    println!("main-workspace/{}: Request review from peers", collection);
-    editor_client.request_review("I made changes").await?;
-
-    print!(
-        "main-workspace/{}: Fetching preview records with default `Verifier`...",
-        collection
-    );
-    let mut preview_client = Client::builder()
-        .http_client(Box::new(ViaductClient))
-        .server_url(server_url)
-        .bucket_name("main-preview")
-        .collection_name(collection)
-        .build()
-        .unwrap();
+    if authorization.is_empty() {
+        println!("⚠️  No AUTHORIZATION env var defined, skipping write operations.");
+    } else {
```

Comment:
Would like to see this broken out into another function, just because I dislike long if's.